controller/node_controller.go

```diff
@@ -536,190 +552,154 @@ func (nc *NodeController) enqueueKubernetesNode(obj interface{}) {
 	nc.enqueueNode(nodeRO)
 }
 
-type diskInfo struct {
-	entry *util.DiskInfo
-	err   error
-}
+func (nc *NodeController) syncDiskStatus(node *longhorn.Node, collectedDataInfo map[string]*monitor.CollectedDiskInfo) error {
+	alignDiskSpecAndStatus(node)
 
-func (nc *NodeController) getDiskInfoMap(node *longhorn.Node) map[string]*diskInfo {
-	result := map[string]*diskInfo{}
-	for id, disk := range node.Spec.Disks {
-		info, err := nc.getDiskInfoHandler(disk.Path)
-		result[id] = &diskInfo{
-			entry: info,
-			err:   err,
-		}
+	notReadyDiskInfoMap, readyDiskInfoMap := nc.fsidDiskGroupMap(node, collectedDataInfo)
+
+	for _, diskInfoMap := range notReadyDiskInfoMap {
+		nc.updateNotReadyDiskStatusReadyCondition(node, diskInfoMap)
 	}
-	return result
-}
 
-// Check all disks in the same filesystem ID are in ready status
-func (nc *NodeController) isFSIDDuplicatedWithExistingReadyDisk(name string, disks []string, diskStatusMap map[string]*longhorn.DiskStatus) bool {
-	if len(disks) > 1 {
-		for _, otherName := range disks {
-			diskReady :=
-				types.GetCondition(
-					diskStatusMap[otherName].Conditions,
-					longhorn.DiskConditionTypeReady)
-
-			if (otherName != name) && (diskReady.Status ==
-				longhorn.ConditionStatusTrue) {
-				return true
-			}
-		}
+	for fsid, diskInfoMap := range readyDiskInfoMap {
+		nc.updateReadyDiskStatusReadyCondition(node, fsid, diskInfoMap)
 	}
 
-	return false
+	return nc.updateDiskStatusSchedulableCondition(node)
 }
 
-func (nc *NodeController) syncDiskStatus(node *longhorn.Node) error {
-	log := getLoggerForNode(nc.logger, node)
+func (nc *NodeController) fsidDiskGroupMap(node *longhorn.Node, collectedDataInfo map[string]*monitor.CollectedDiskInfo) (notReadyDiskInfoMap, readyDiskInfoMap map[string]map[string]*monitor.CollectedDiskInfo) {
```

Comment:
Sorry, I mean renaming the below variable rather than this function... My bad...

Reply:
That's okay. But maybe `fsidDiskToGroupMap` is also a good name for this function.